src/main/java/net/revelc/code/impsort/LineEnding.java

```diff
@@ -43,10 +43,11 @@ public static LineEnding determineLineEnding(String fileDataString) {
     int crCount = 0;
     int crlfCount = 0;
 
-    for (int i = 0; i < fileDataString.length(); i++) {
-      char c = fileDataString.charAt(i);
+    final int length = fileDataString.length();
+    for (int i = 0; i < length; i++) {
+      final char c = fileDataString.charAt(i);
       if (c == '\r') {
-        if ((i + 1) < fileDataString.length() && fileDataString.charAt(i + 1) == '\n') {
+        if ((i + 1) < length && fileDataString.charAt(i + 1) == '\n') {
```

Comment:
While this looks better on the surface, there's really no performance impact to calling `String.length()`. Since this code is copied from another source, I'd prefer to keep it unmodified, to keep it in sync with its original source, unless there's significant value to the change. I don't think that's the case here, as this saves us from one method call, and one bitwise shift operation. Also, since this is an unrelated change to what this patch is intended to fix, it'd be better to keep the diff small, so it's easier to analyze when looking back through the git history, rather than include extra unrelated changes.

Reply:
OK, I will revert this. At some point, I had probably changed something here related to the PR, but it no longer exists.